metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -251,7 +267,7 @@ public void deactivate() {
 
         List<ApiMessageAndVersion> records = new ArrayList<>();
         records.add(new ApiMessageAndVersion(record,
-            REGISTER_BROKER_RECORD.highestSupportedVersion()));
+            metadataVersionProvider.activeVersion().recordVersion(REGISTER_BROKER_RECORD)));
```

Comment:
Assuming we are fine with this usage pattern, we will need to make this change any time we serialize a record